src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -31,4 +149,53 @@ mdAssemblyRef FindAssemblyRef(
   return mdAssemblyRefNil;
 }
 
+std::vector<Integration> FilterIntegrationsByCaller(
+    const std::vector<Integration>& integrations,
+    const std::wstring& assembly_name) {
+  std::vector<Integration> enabled;
+
+  for (auto& i : integrations) {
+    bool found = false;
+    for (auto& mr : i.method_replacements) {
+      if (mr.caller_method.assembly.name.empty() ||
+          mr.caller_method.assembly.name == assembly_name) {
+        found = true;
+      }
+    }
+    if (found) {
+      enabled.push_back(i);
+    }
+  }
+
+  return enabled;
+}
+
+std::vector<Integration> FilterIntegrationsByTarget(
+    const std::vector<Integration>& integrations,
+    const ComPtr<IMetaDataAssemblyImport>& assembly_import) {
+  std::vector<Integration> enabled;
+
+  auto assembly_name = GetAssemblyName(assembly_import);
+
+  for (auto& i : integrations) {
+    bool found = false;
+    for (auto& mr : i.method_replacements) {
+      if (mr.target_method.assembly.name == assembly_name) {
+        found = true;
```

Comment:
can we break out early out of these loops when we find the first match? looks like in both loops we can replace
```c++
found = true;
```
with
```c++
enabled.push_back(i);
break;
```